### PR TITLE
Fix error handling

### PR DESCRIPTION
**Pull request by yaaax**
Catch error in rejected promises

Fixes infinite pending modal on timeout

### CODE REVIEW

**Comment by yaaax** [comment]
@vvatelot I think we should merge that one, I still see a lot of cases where the modal for the EcoIndex analysis stays on screen indefinitely

**Comment by jpkha** [comment]
Who can validate and merge the PR ? @yaaax  @vvatelot

File: assets/js/services/AnalysisService.js [hunk 1]
```diff
@@ -22,26 +22,35 @@ class AnalysisService {
...
-					}
-				})
-			});
+							if (taskResult.status === "FAILURE") {
+								EcoIndexDialog.openErrorMessage(599, taskResult.task_error);
+							}
+						},
+						(e) => {
+							this.#handleError(e);
+						}
```
Comment (vvatelot) [inline assets/js/services/AnalysisService.js:47]: **❓ question(non-blocking):** What if the API responds with a 425 on `fetchAnalysisTaskById` ? :thinking:

Comment (vvatelot) [reply assets/js/services/AnalysisService.js:47]: Seems that it works well ! :)

**Review by vvatelot — APPROVED** [review]

**Pull request merged**